src/native/corehost/hostmisc/pal.windows.cpp

```diff
@@ -276,8 +276,10 @@ bool pal::get_default_installation_dir(pal::string_t* recv)
     }
     //  ***************************
 
+    USHORT pProcessMachine, pNativeMachine;
     const pal::char_t* program_files_dir;
-    if (pal::is_running_in_wow64())
+    auto is_running_in_wow64 = IsWow64Process2(GetCurrentProcess(), &pProcessMachine, &pNativeMachine);
```

Comment:
`IsWow64Process2` is only present in more recent Win10. Anything older than is missing this API.

You need to get it via LoadLibraryEx / GetProcAddress, and fallback to what was here before if the API is not present.